src/test/java/mate/academy/bookstore/services/CategoryServiceTest.java

```diff
@@ -0,0 +1,241 @@
+            """)
+    @Test
+    void createCategory_WithValidRequestDto_ShouldReturnCategoryResponseDto() {
+        when(categoryMapper.toModel(categoryRequestDto)).thenReturn(category);
+        when(categoryRepository.save(category)).thenReturn(category);
+        when(categoryMapper.toCategoryDto(category)).thenReturn(categoryResponseDto);
+
+        CategoryResponseDto actual = categoryService.save(categoryRequestDto);
+
+        assertNotNull(actual);
+        assertEquals(TEST_CATEGORY_ID, actual.getId());
+
+        verify(categoryMapper, times(1)).toModel(categoryRequestDto);
+        verify(categoryRepository, times(1)).save(category);
+        verify(categoryMapper, times(1)).toCategoryDto(category);
+        verifyNoMoreInteractions(categoryRepository, categoryMapper);
+    }
+
+    @DisplayName("""
+        Find all categories with a valid request
+        Should return Page list of CategoryResponseDto
+            """)
+    @Test
+    void findAllCategories_WithValidRequestDto_ShouldReturnCategoriesResponseDto() {
+        Pageable pageable = PageRequest.of(0, 10);
+        Page<Category> categoryPage = new PageImpl<Category>(categories, pageable,
+                categories.size());
+
+        when(categoryRepository.findAll(pageable)).thenReturn(categoryPage);
+
+        List<CategoryResponseDto> expectedBookDtos = categories.stream()
+                .map(categoryMapper::toCategoryDto)
+                .toList();
+
+        Page<CategoryResponseDto> actual = categoryService.findAll(pageable);
+
+        assertNotNull(actual);
+        assertEquals(expectedBookDtos.size(), actual.getContent().size());
+    }
+
+    @DisplayName("""
+        Get category by ID with valid ID
+        Should return the CategoryResponseDto
+            """)
+    @Test
+    void getCategoryById_WithValidId_ShouldReturnCategoriesResponseDto() {
+        when(categoryRepository.findById(TEST_CATEGORY_ID)).thenReturn(Optional.of(category));
+        when(categoryMapper.toCategoryDto(category)).thenReturn(categoryResponseDto);
+
+        CategoryResponseDto actual = categoryService.getById(TEST_CATEGORY_ID);
+
+        assertNotNull(actual);
+        assertEquals(TEST_CATEGORY_ID, actual.getId());
+    }
+
+    @DisplayName("""
+        Get category by ID with non-existing ID
+        Should throw EntityNotFoundException
+            """)
+    @Test
+    void getCategoryById_WithNonExistingId_ShouldThrowException() {
+        when(categoryRepository.findById(TEST_INCORRECT_CATEGORY_ID)).thenReturn(Optional.empty());
+
+        EntityNotFoundException entityNotFoundException =
+                assertThrows(EntityNotFoundException.class,
+                    () -> categoryService.getById(TEST_INCORRECT_CATEGORY_ID));
+
+        String expectedMessage = "Can't find category by id " + TEST_INCORRECT_CATEGORY_ID;
+        String actualMessage = entityNotFoundException.getMessage();
+
+        assertEquals(expectedMessage, actualMessage);
```

Comment:
```suggestion
        String actualMessage = entityNotFoundException.getMessage();

        assertTrue(actualMessage.contains(String.valueOf(TEST_INCORRECT_CATEGORY_ID)));
```
we can make it less dependent on specific wording while still being useful

Reply:
nice idea